app/models/course/condition.rb

```diff
@@ -3,4 +3,6 @@ class Course::Condition < ActiveRecord::Base
 
   belongs_to :course, inverse_of: false
   belongs_to :conditional, polymorphic: true
+
+  delegate :satisfied_by?, to: :actable
```

Comment:
But that abstract implementation in base.rb:46 is like a protocol, it's not a superclass of specific conditions. How to delegate it to derived implementation? What I'm thinking is: from a conditional, we access `Course::Condition` via for example `achievement.conditions` and then each `Course::Condition` delegates to its specific condition.